internal/tools/aws/database_multitenant.go

```diff
@@ -0,0 +1,717 @@
+		SecretId: lockedCluster.cluster.DBClusterIdentifier,
+	})
+	if err != nil {
+		return errors.Wrapf(err, "unable to find the master secret for the multitenant database ID %s", *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	close, err := d.connectRDSCluster(rdsMySQLSchemaInformationDatabase, *lockedCluster.cluster.Endpoint, DefaultMattermostDatabaseUsername, *masterSecretValue.SecretString)
+	if err != nil {
+		return errors.Wrapf(err, "unable to connect to the multitenant RDS cluster ID %s", *lockedCluster.cluster.DBClusterIdentifier)
+	}
+	defer close(logger)
+
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(DefaultMySQLContextTimeSeconds*time.Second))
+	defer cancel()
+
+	err = d.createDatabaseIfNotExist(ctx, installationDatabaseName)
+	if err != nil {
+		return errors.Wrapf(err, "unable to create schema in multitenant RDS cluster ID %s", *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	installationSecret, err := d.ensureMultitenantDatabaseSecretIsCreated(lockedCluster.cluster.DBClusterIdentifier, vpc.VpcId)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get a database secrete for installation ID %s", d.installationID)
+	}
+
+	err = d.createUserIfNotExist(ctx, installationSecret.MasterUsername, installationSecret.MasterPassword)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create a Mattermost database schema for installation ID %s", d.installationID)
+	}
+
+	err = d.grantUserFullPermissions(ctx, installationDatabaseName, installationSecret.MasterUsername)
+	if err != nil {
+		return errors.Wrapf(err, "failed to grant permissions to Mattermost database user for installation ID %s", d.installationID)
+	}
+
+	databaseInstallationIDs, err := store.AddMultitenantDatabaseInstallationID(*lockedCluster.cluster.DBClusterIdentifier, d.installationID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to add the installation ID %s to multitenant database ID %s", d.installationID, *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	err = d.updateCounterTag(lockedCluster.cluster.DBClusterArn, len(databaseInstallationIDs))
+	if err != nil {
+		return errors.Wrapf(err, "failed to update tag:counter in RDS cluster ID %s", *lockedCluster.cluster.DBClusterIdentifier)
+	}
+
+	logger.Infof("Multitenant database ID %s has %d installations", *lockedCluster.cluster.DBClusterIdentifier, len(databaseInstallationIDs))
+
+	return nil
+}
+
+// Helpers
+
+type validatedAndLockedRDSCluster struct {
+	unlock  func(log.FieldLogger)
+	cluster *rds.DBCluster
+}
+
+func (d *RDSMultitenantDatabase) validateAndLockRDSCluster(multitenantDatabases []*model.MultitenantDatabase, store model.InstallationDatabaseStoreInterface, logger log.FieldLogger) (*validatedAndLockedRDSCluster, error) {
+	for _, multitenantDatabase := range multitenantDatabases {
+		installations, err := multitenantDatabase.GetInstallationIDs()
```

Comment:
It would be really nice if the body of this `for` loop was broken into a helper method

Reply:
Is this for testing reasons, readability, or something else?

Reply:
Ok, I see the reason in the comment below. I will do the change.